# Project profitable App Profiles

This project, we'll pretend we're working as data analysts for a company that builds Android and iOS mobile apps for an English-speaking audience. 
We make our apps available on Google Play and in the App Store and the revenue comes from in-app ads.
The goal of this project is to analyze data to help our developers understand what type of apps are likely to attract more users.

## Open and explore data

In [2]:
from csv import reader


opened_gg_file = open('googleplaystore.csv')
read_gg_file = reader(opened_gg_file)
android = list (read_gg_file)
android_header = android[0]
android = android[1:]

opened_apple_file = open('AppleStore.csv',encoding='utf8')
read_apple_file = reader(opened_apple_file)
iOs = list (read_apple_file)
iOs_header = iOs[0]
iOs = iOs[1:]
    

In [3]:
android_header

['App',
 'Category',
 'Rating',
 'Reviews',
 'Size',
 'Installs',
 'Type',
 'Price',
 'Content Rating',
 'Genres',
 'Last Updated',
 'Current Ver',
 'Android Ver']

In [4]:
iOs_header

['id',
 'track_name',
 'size_bytes',
 'currency',
 'price',
 'rating_count_tot',
 'rating_count_ver',
 'user_rating',
 'user_rating_ver',
 'ver',
 'cont_rating',
 'prime_genre',
 'sup_devices.num',
 'ipadSc_urls.num',
 'lang.num',
 'vpp_lic']

In [5]:
def explore_data(dataset,start,end,rows_and_columns=False):
    '''
    This function is used to print first few row of data set for view and 
        return the number of row and column in the data set
    Parameter:
        dataset: a list of lists
        start,end: number
        rows_and_columns: condition identifier
    '''
    dataset_slice = dataset[start:end]
    for row in dataset_slice:
        print(row)
        print('\n')
    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns', len(dataset[0]))

In [6]:
explore_data(android, 0,10,rows_and_columns=True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


['Paper flowers instructions', 'ART_AND_DESIGN', '4.4', '167', '5.6M', '50,000+

In [7]:
explore_data(iOs, 0,10,rows_and_columns=True)

['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


['284035177', 'Pandora - Music & Radio', '130242560', 'USD', '0.0', '1126879', '3594', '4.0', '4.5', '8.4.1', '12+', 'Music', '37', '4', '1', '1']


['429047995', 'Pinterest', '74778624', 'USD', '0.0', '1061624', '1814', '4.5', '4.0', '6.26', '12+', 'Social Networking', '37', '5', '27', '1']


['282935706', 'Bible', '92774400', 'USD', '0.0', '985920', '5320', '4.5', '5.0', '7.5.1', '4+', 'Reference', '37', '5', '45', '1']


['5538347

## Clean data

### Remove uncleaned data
+ Detect inaccurate data, and correct or remove it.
+ Detect duplicate data, and remove the duplicates

#### Found row that misses data and delete it

In [8]:
for index,row in enumerate(android):
    if len(row) != len(android_header):
        print(row)
        print(index)

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']
10472


In [9]:
del android[10472]

In [10]:
for index,row in enumerate(iOs):
    if len(row) != len(iOs_header):
        print(row)
        print(index)

There is no such error in AppleStore data

#### Remove duplicates entries

In [11]:
#initiate lists to collect duplicated apps and store apps name distinctly
android_duplicate_apps = []
android_unique_apps = []

for app in android:
    if app[0] in android_unique_apps:
        android_duplicate_apps.append(app[0])
    else:
        android_unique_apps.append(app[0])
for name in android_duplicate_apps[5:8]:
    for app in android:
        if app[0] == name:
            print(app)

['Box', 'BUSINESS', '4.2', '159872', 'Varies with device', '10,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Box', 'BUSINESS', '4.2', '159872', 'Varies with device', '10,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Box', 'BUSINESS', '4.2', '159872', 'Varies with device', '10,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Zenefits', 'BUSINESS', '4.2', '296', '14M', '50,000+', 'Free', '0', 'Everyone', 'Business', 'June 15, 2018', '3.2.1', '4.1 and up']
['Zenefits', 'BUSINESS', '4.2', '296', '14M', '50,000+', 'Free', '0', 'Everyone', 'Business', 'June 15, 2018', '3.2.1', '4.1 and up']
['Google Ads', 'BUSINESS', '4.3', '29313', '20M', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 30, 2018', '1.12.0', '4.0.3 and up']
['Google Ads', 'BUSINESS', '4.3', '29313', '20M', '5,000,000+', 'Free', 

In [12]:
len(android_duplicate_apps)

1181

In [13]:
#to remove duplicate, we need to create a dictionary to store max reviews per app since that would be an identifired for latest data
reviews_max = {}

for app in android:
    name = app[0]
    n_reviews = float(app[3])
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
    elif name not in reviews_max:
        reviews_max[name] = n_reviews

In [14]:
len(reviews_max)

9659

In [15]:
android_clean = []
already_added = []

for app in android:
    name = app[0]
    n_reviews = float(app[3])
    if name not in already_added and n_reviews == reviews_max[name]:
        android_clean.append(app)
        already_added.append(name)

In [16]:
len(android_clean)

9659

In [17]:
#initiate lists to collect duplicated apps and store apps name distinctly
iOs_duplicate_apps = []
iOs_unique_apps = []

for app in iOs:
    if app[1] in iOs_unique_apps:
        iOs_duplicate_apps.append(app[1])
    else:
        iOs_unique_apps.append(app[1])
for name in iOs_duplicate_apps[:8]:
    for app in iOs:
        if app[1] == name:
            print(app)

['1173990889', 'Mannequin Challenge', '109705216', 'USD', '0.0', '668', '87', '3.0', '3.0', '1.4', '9+', 'Games', '37', '4', '1', '1']
['1178454060', 'Mannequin Challenge', '59572224', 'USD', '0.0', '105', '58', '4.0', '4.5', '1.0.1', '4+', 'Games', '38', '5', '1', '1']
['952877179', 'VR Roller Coaster', '169523200', 'USD', '0.0', '107', '102', '3.5', '3.5', '2.0.0', '4+', 'Games', '37', '5', '1', '1']
['1089824278', 'VR Roller Coaster', '240964608', 'USD', '0.0', '67', '44', '3.5', '4.0', '0.81', '4+', 'Games', '38', '0', '1', '1']


In [18]:
# clean iOs using the same approach with how we clean android data
iOs_reviews_max = {}

for app in iOs:
    name = app[1]
    n_reviews = float(app[5])
    if name in iOs_reviews_max and iOs_reviews_max[name] < n_reviews:
        iOs_reviews_max[name] = n_reviews
    elif name not in iOs_reviews_max:
        iOs_reviews_max[name] = n_reviews
        
iOs_clean = []
iOs_already_added = []

for app in iOs:
    name = app[1]
    n_reviews = float(app[5])
    if name not in iOs_already_added and n_reviews == iOs_reviews_max[name]:
        iOs_clean.append(app)
        iOs_already_added.append(name)

In [19]:
len(iOs_clean)

7195

Recall that at our company, we only build apps that are free to download and install, and we design them for an English-speaking audience. This means that we'll need to do the following:

#### Remove non-English apps like 爱奇艺PPS -《欢乐颂2》电视剧热播.

In [20]:
def is_english(string):
    '''
    The numbers corresponding to the characters we commonly use in an English text are all in the range 0 to 127
    We loop each character in the string to identify any character that is non-english.
    If the input string has more than three characters that fall outside the ASCII range (0 - 127), that string is non-english
    '''
    non_english_characters = 0
    for character in string:
        if ord(character) > 127:
            non_english_characters +=1
        if non_english_characters > 3:
            return False
    return True

print(is_english('Docs To Go™ Free Office Suite'))
print('\n')
print(is_english('Instachat 😜'))
print('\n')
print(is_english('爱奇艺PPS -《欢乐颂2》电视剧热播'))

True


True


False


In [21]:
english_android_app = []
non_english_android_app = []
for app in android_clean:
    name = app[0]
    if is_english(name):
        english_android_app.append(app)
    if not is_english(name):
        non_english_android_app.append(app)


In [22]:
len(english_android_app)

9614

In [23]:
english_iOs_app = []
non_english_iOs_app = []
for app in iOs_clean:
    name = app[1]
    if is_english(name):
        english_iOs_app.append(app)
    if not is_english(name):
        non_english_iOs_app.append(app)

In [24]:
len(english_iOs_app)

6181

#### Isolate apps that aren't free.

In [25]:
free_english_android_app = []
for app in english_android_app:
    if app[6] !='Paid':
        free_english_android_app.append(app)

In [26]:
len(free_english_android_app)

8864

In [27]:
free_english_iOS_app = []
for app in english_iOs_app:
    if app[4] !='0.0':
        free_english_iOS_app.append(app)

In [28]:
len(free_english_iOS_app)

2961

## Analyze data

Because our end goal is to add the app on both Google Play and the App Store, we need to find app profiles that are successful in both markets. For instance, a profile that works well for both markets might be a productivity app that makes use of gamification.

### Find the most common apps by genre

In [29]:
## create a function to generate frequency tables that show percentages

def freq_table(dataset, index):
    table = {}
    total = 0

    for row in dataset:
        total+=1
        value = row[index]
        if value in table:
            table[value]+=1
        else:
            table[value]=1
            
    table_percentage = {}
    
    for key in table:
        percentage = (table[key]/total) *100
        table_percentage[key] = percentage
        
    return table_percentage
            

In [30]:
## create a function to sort table by percent in descending order
def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

In [31]:
display_table(free_english_iOS_app, 11)

Games : 51.266464032421474
Education : 9.861533265788584
Entertainment : 6.5856129685916915
Photo & Video : 6.112799729821006
Utilities : 4.4579533941236065
Productivity : 3.7825059101654848
Health & Fitness : 3.3772374197906116
Music : 2.397838568051334
Lifestyle : 1.6210739614994936
Weather : 1.3846673421141507
Book : 1.3846673421141507
Business : 1.21580547112462
Sports : 1.1820330969267139
Reference : 1.1820330969267139
Navigation : 0.7429922323539345
Travel : 0.6754474839581223
Social Networking : 0.6754474839581223
Food & Drink : 0.60790273556231
Medical : 0.5065856129685917
News : 0.4728132387706856
Finance : 0.4390408645727794
Shopping : 0.033772374197906116
Catalogs : 0.033772374197906116


We can see that among the free English apps, more than a half (58.16%) are games. Entertainment apps are close to 8%, followed by photo and video apps, which are close to 5%. Only 3.66% of the apps are designed for education, followed by social networking apps which amount for 3.29% of the apps in our data set.

The general impression at least taken from free english apps in App store are most of the apps designed for entertainment (games, photo and video, social networking, sports, music) rather than for practical purpose  (education, shopping, utilities, productivity, lifestyle).

However, we can't conclude that the same pattern would apply for user taste, a large number of apps for a particular genre doesn't imply that apps of that genre generally have a large number of users.

Let's examine the trend on Google play by looking into app's genre and category.

In [32]:
list_android = [1,9]
for column in list_android:
    display_table(free_english_android_app, column)
    print('\n')



FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

The landscape seems significantly different on Google Play: there are not that many apps designed for fun, and it seems that a good number of apps are designed for practical purposes (family, tools, business, lifestyle, productivity, etc.). This picture is also confirmed by the frequency table we see for the Genres column.

The difference between the Genres and the Category columns is not crystal clear, but one thing we can notice is that the Genres column is much more granular (it has more categories). We're only looking for the bigger picture at the moment, so we'll only work with the Category column moving forward.

### Find the most common apps by genre in App Store

The frequency tables we analyzed on the previous screen showed us that apps designed for fun dominate the App Store, while Google Play shows a more balanced landscape of both practical and fun apps. Now, we'd like to determine the kind of apps with the most users.

One way to find out what genres are the most popular (have the most users) is to calculate the average number of installs for each app genre. For the Google Play data set, we can find this information in the Installs column, but this information is missing for the App Store data set. As a workaround, we'll take the total number of user ratings as a proxy, which we can find in the `rating_count_tot` app.

In [33]:
#calculate the average number of user ratings for each genre
#generate frequecy table for prime_genre
prime_genre =freq_table(free_english_iOS_app, -5)

for genre in prime_genre:
    total = 0
    len_genre = 0
    for app in free_english_iOS_app:
        genre_app = app[-5]
        user_ratings = float(app[5])
        if genre_app == genre:
            total+= user_ratings
            len_genre+= 1
    average_n_rating = total/len_genre
    print(f'{genre}: {average_n_rating}')
            

Games: 6695.863636363636
Entertainment: 2131.5128205128203
Music: 2759.1971830985917
Photo & Video: 2531.519337016575
Health & Fitness: 2679.85
Business: 4043.472222222222
Weather: 3248.4146341463415
Utilities: 1326.6818181818182
News: 3872.3571428571427
Education: 640.972602739726
Reference: 2400.3714285714286
Productivity: 2247.9285714285716
Navigation: 1174.590909090909
Lifestyle: 902.7708333333334
Book: 320.4146341463415
Finance: 882.8461538461538
Sports: 253.74285714285713
Medical: 663.7333333333333
Travel: 602.95
Shopping: 2722.0
Food & Drink: 579.5
Social Networking: 393.0
Catalogs: 1309.0


On average, Games apps still have the highest number of user reviews, but this figure is heavily influenced by Fruit Ninja Classic, which have more than half a million user reviews together:

In [34]:
for app in free_english_iOS_app:
    if app[-5] == 'Games':
        print(app[1], ':', app[5]) # print name and number of ratings

Fruit Ninja Classic : 698516
Clear Vision (17+) : 541693
Minecraft: Pocket Edition : 522012
Plants vs. Zombies : 426463
Doodle Jump : 395261
Draw Something : 360974
Infinity Blade : 326482
Geometry Dash : 266440
Tiny Wings : 219418
Traffic Rush : 213092
Plants vs. Zombies HD : 163598
Infinity Blade II : 153588
The Room : 143908
Plague Inc. : 143285
Where's My Water? : 131656
Call of Duty: Black Ops Zombies : 116601
Sunday Lawn : 107727
SCRABBLE Premium : 105776
Flick Home Run ! : 98851
Terraria : 98036
Scribblenauts Remix : 86127
Zombieville USA 2 : 79683
Trivia Crack (No Ads) : 77370
Solitaire by MobilityWare : 76720
Call of Duty: Zombies : 63943
Cut the Rope: Experiments : 63272
Phase 10 Pro - Play Your Friends! : 59155
The Sims 3 : 54408
HB2 PLUS : 54073
StickWars : 53821
BATTLE BEARS: Zombies! : 50710
Slayin : 45574
Ski Safari : 45121
Lane Splitter : 44567
Bloons TD 5 : 42078
Fieldrunners : 41633
Super Stickman Golf : 41446
Cat Physics : 40552
SCRABBLE Premium for iPad : 39678
Infi

Business apps on the other hands are denomiated by Scanner, PDF reader apps and there are already lots of app out there so this is not a niche area.

In [35]:
for app in free_english_iOS_app:
    if app[-5] == 'Business':
        print(app[1], ':', app[5]) # print name and number of ratings

Scanner Pro - PDF document scanner app with OCR : 31912
Splashtop Personal - Remote Desktop : 29376
TurboScan™ Pro - document & receipt scanner: scan multiple pages and photos to PDF : 28388
iScanner - PDF Document Scanner App : 8675
PDF Reader Pro Edition - Annotate,edit & sign PDFs : 7033
InstaLogo Logo Creator - Graphic design maker : 6263
Tiny Scanner+ - PDF scanner to scan document, receipt & fax : 5611
HotSchedules : 3292
ScanBizCards Business Card Reader : 3166
TapeACall Pro - Call Recorder For Phone Calls : 2951
CamCard - Business card scanner & reader : 2923
File Manager Pro App : 2154
Scanner For Me - PDF Scan with OCR for Documents : 1976
AudioNote - Notepad and Voice Recorder : 1756
SuperCam_Pro : 1399
PDF Converter - Convert Documents, Photos to PDF : 1294
Voice Translate Pro : 1158
Fax from iPhone - send fax app : 1039
Genius Scan+ - PDF Scanner : 916
OfficeSuite Pro (Mobile Office) : 896
Consultant Aide : 881
SamCard Pro-card reader&business card scanner&ocr : 875
ALON D

We look at the 3rd genre in average number of user reviews. They are mostly podcast app which requires human content and is not ideal to build merely by code.

In [36]:
for app in free_english_iOS_app:
    if app[-5] == 'News':
        print(app[1], ':', app[5]) # print name and number of ratingsNews

5-0 Radio Pro Police Scanner (Extra Feeds) : 24553
Scanner911 Pro : 14345
Daily Classifieds for iPhone : 7973
Conservative Talk Radio : 2436
PodCruncher podcast app - Player and manager for podcasts : 2261
Pocket Casts : 1911
News Pro - Breitbart Edition : 244
Presidential Election & Electoral College Maps : 222
Scanner911 HD Police Radio Pro : 204
PS Deals+ - Games Price Alerts for PS4, PS3, Vita : 28
Electoral 2016 - Create Presidential Election Maps : 22
Boycott Trump Biz : 14
ZAKER 专业版 : 0
Verbrechen - echte Polizeifälle aus Deiner Umgebung : 0


Reference apps is the next runner in user ratings on average, after we exclude Music, Photo & Video, Health & Fitness since these genre requires the same effort of human content as News

In [37]:
for app in free_english_iOS_app:
    if app[-5] == 'Reference':
        print(app[1], ':', app[5])

Sky Guide: View Stars Night or Day : 22100
Dictionary.com Dictionary & Thesaurus Premium : 11530
WolframAlpha : 7410
e-Sword HD: Bible Study Made Easy : 7309
iHandy Translator Pro : 5163
Dictionary.com Premium Dictionary & Thesaurus for iPad : 4922
Speak & Translate － Live Voice and Text Translator : 4344
National Geographic World Atlas : 4255
Knots 3D : 3196
iQuran : 2929
Merriam-Webster Dictionary & Thesaurus : 2843
e-Sword LT: Bible Study on the Go : 2152
Barefoot World Atlas : 1161
Catholic Bible : 999
CCW – Concealed Carry 50 State Guide : 644
Voice Translator with Offline Dictionary Pro : 499
DINOSAUR MOD FOR MINECRAFT PC EDITION - MODS POCKET GUIDE : 413
Mutant Creatures Mod for Minecraft PC Edition - Pocket Mods Guide : 346
Fighter Verses - memorize bible verses : 331
DRAGON MODS MINE EDITION FOR MINECRAFT GAME PC : 318
The World Factbook for iPad : 273
Catholic New American Bible Revised Edition : 149
V for Wikipedia : 134
SkySafari 5 Plus : 108
Great App for Who is Your Daddy

This niche seems to show some potential. One thing we could do is take another popular book and turn it into an app where we could add different features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes about the book, etc. On top of that, we could also embed a dictionary within the app, so users don't need to exit our app to look up words in an external app.

This idea seems to fit well with the fact that the App Store is dominated by for-fun apps. This suggests the market might be a bit saturated with for-fun apps, which means a practical app might have more of a chance to stand out among the huge number of apps on the App Store.

Other genres that seem popular include weather, book, food and drink, or finance. The book genre seem to overlap a bit with the app idea we described above, but the other genres don't seem too interesting to us:

+ Weather apps — people generally don't spend too much time in-app, and the chances of making profit from in-app adds are low. Also, getting reliable live weather data may require us to connect our apps to non-free APIs.

+ Food and drink — examples here include Starbucks, Dunkin' Donuts, McDonald's, etc. So making a popular food and drink app requires actual cooking and a delivery service, which is outside the scope of our company.

+ Finance apps — these apps involve banking, paying bills, money transfer, etc. Building a finance app requires domain knowledge, and we don't want to hire a finance expert just to build an app.

Now let's analyze the Google Play market a bit.

### Find the most common apps by genre in Google Play

In [38]:
display_table(free_english_android_app,5) #Install

1,000,000+ : 15.726534296028879
100,000+ : 11.552346570397113
10,000,000+ : 10.548285198555957
10,000+ : 10.198555956678701
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.7721119133574
5,000+ : 4.512635379061372
10+ : 3.5424187725631766
500+ : 3.2490974729241873
50,000,000+ : 2.3014440433213
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.78971119133574
1+ : 0.5076714801444043
500,000,000+ : 0.2707581227436823
1,000,000,000+ : 0.22563176895306858
0+ : 0.04512635379061372
0 : 0.01128158844765343


We're going to leave the numbers as they are, which means that we'll consider that an app with 100,000+ installs has 100,000 installs, and an app with 1,000,000+ installs has 1,000,000 installs, and so on. To perform computations, however, we'll need to convert each install number from a string to a float. This means we need to remove the commas and the plus characters, or the conversion will fail and cause an error.

In [39]:
categories_android = freq_table(free_english_android_app, 1)
best_category = ''
highest_n_installs = 0

for category in categories_android:
    total = 0
    len_category = 0
    for app in free_english_android_app:
        category_app = app[1]
        if category_app == category:            
            n_installs = app[5]
            n_installs = n_installs.replace(',', '')
            n_installs = n_installs.replace('+', '')
            total += float(n_installs)
            len_category += 1
    avg_n_installs = total / len_category
    
    if avg_n_installs > highest_n_installs:
        best_category = category
        highest_n_installs = avg_n_installs
    print(category, ':', avg_n_installs)
print(f'Best category: {best_category}')
print(f'Number of installs: {highest_n_installs}')

ART_AND_DESIGN : 1986335.0877192982
AUTO_AND_VEHICLES : 647317.8170731707
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 8767811.894736841
BUSINESS : 1712290.1474201474
COMICS : 817657.2727272727
COMMUNICATION : 38456119.167247385
DATING : 854028.8303030303
EDUCATION : 1833495.145631068
ENTERTAINMENT : 11640705.88235294
EVENTS : 253542.22222222222
FINANCE : 1387692.475609756
FOOD_AND_DRINK : 1924897.7363636363
HEALTH_AND_FITNESS : 4188821.9853479853
HOUSE_AND_HOME : 1331540.5616438356
LIBRARIES_AND_DEMO : 638503.734939759
LIFESTYLE : 1437816.2687861272
GAME : 15588015.603248259
FAMILY : 3695641.8198090694
MEDICAL : 120550.61980830671
SOCIAL : 23253652.127118643
SHOPPING : 7036877.311557789
PHOTOGRAPHY : 17840110.40229885
SPORTS : 3638640.1428571427
TRAVEL_AND_LOCAL : 13984077.710144928
TOOLS : 10801391.298666667
PERSONALIZATION : 5201482.6122448975
PRODUCTIVITY : 16787331.344927534
PARENTING : 542603.6206896552
WEATHER : 5074486.197183099
VIDEO_PLAYERS : 24727872.452830188
NEWS_AND_

On average, communication apps have the most installs: 38,456,119. This number is heavily skewed up by a few apps that have over one billion installs (WhatsApp, Facebook Messenger, Skype, Google Chrome, Gmail, and Hangouts), and a few others with over 100 and 500 million installs:

In [40]:
for app in free_english_android_app:
    if app[1] == 'COMMUNICATION':
        print(f'{app[0]} : {app[5]}')

WhatsApp Messenger : 1,000,000,000+
Messenger for SMS : 10,000,000+
My Tele2 : 5,000,000+
imo beta free calls and text : 100,000,000+
Contacts : 50,000,000+
Call Free – Free Call : 5,000,000+
Web Browser & Explorer : 5,000,000+
Browser 4G : 10,000,000+
MegaFon Dashboard : 10,000,000+
ZenUI Dialer & Contacts : 10,000,000+
Cricket Visual Voicemail : 10,000,000+
TracFone My Account : 1,000,000+
Xperia Link™ : 10,000,000+
TouchPal Keyboard - Fun Emoji & Android Keyboard : 10,000,000+
Skype Lite - Free Video Call & Chat : 5,000,000+
My magenta : 1,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Seznam.cz : 1,000,000+
Antillean Gold Telegram (original version) : 100,000+
AT&T Visual Voicemail : 10,000,000+
GMX Mail : 10,000,000+
Omlet Chat : 10,000,000+
My Vodacom SA : 5,000,000+
Microsoft Edge : 5,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Calls & Text by Mo+ : 5,000,000+
free 

If we removed all the communication apps that have over 100 million installs, the average would be reduced roughly ten times:

In [41]:
under_100_m = []

for app in free_english_android_app:
    n_installs = app[5]
    n_installs = n_installs.replace(',', '')
    n_installs = n_installs.replace('+', '')
    if (app[1] == 'COMMUNICATION') and (float(n_installs) < 100000000):
        under_100_m.append(float(n_installs))
        
sum(under_100_m) / len(under_100_m)

3603485.3884615386

We see the same pattern for the video players category, which is the runner-up with 24,727,872 installs. The market is dominated by apps like Youtube, Google Play Movies & TV, or MX Player. The pattern is repeated for social apps (where we have giants like Facebook, Instagram, Google+, etc.), photography apps (Google Photos and other popular photo editors), or productivity apps (Microsoft Word, Dropbox, Google Calendar, Evernote, etc.).

Again, the main concern is that these app genres might seem more popular than they really are. Moreover, these niches seem to be dominated by a few giants who are hard to compete against.

The game genre seems pretty popular, but previously we found out this part of the market seems a bit saturated, so we'd like to come up with a different app recommendation if possible.

The books and reference genre looks fairly popular as well, with an average number of installs of 8,767,811. It's interesting to explore this in more depth, since we found this genre has some potential to work well on the App Store, and our aim is to recommend an app genre that shows potential for being profitable on both the App Store and Google Play.

Let's take a look at some of the apps from this genre and their number of installs:

In [43]:
for app in free_english_android_app:
    if app[1] == 'BOOKS_AND_REFERENCE':
        print(app[0], ':', app[5])

E-Book Read - Read Book for free : 50,000+
Download free book with green book : 100,000+
Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Free Panda Radio Music : 100,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
English Grammar Complete Handbook : 500,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
Google Play Books : 1,000,000,000+
AlReader -any text book reader : 5,000,000+
Offline English Dictionary : 100,000+
Offline: English to Tagalog Dictionary : 500,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
Recipes of Prophetic Medicine for free : 500,000+
ReadEra – free ebook reader : 1,000,000+
Anonymous caller detection : 10,000+
Ebook Reader : 5,000,000+
Litnet - E-books : 100,000+
Read books online : 5,000,000+
English to Urdu Dictionary : 500,000+
eBoox: book reader fb2 epub zip : 1,000,000+
English Persian Dictionary : 500,000+
Flybook : 500,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
E

The book and reference genre includes a variety of apps: software for processing and reading ebooks, various collections of libraries, dictionaries, tutorials on programming or languages, etc. It seems there's still a small number of extremely popular apps that skew the average:

In [45]:
for app in free_english_android_app:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000,000+'
                                            or app[5] == '500,000,000+'
                                            or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Google Play Books : 1,000,000,000+
Bible : 100,000,000+
Amazon Kindle : 100,000,000+
Wattpad 📖 Free Books : 100,000,000+
Audiobooks from Audible : 100,000,000+


However, it looks like there are only a few very popular apps, so this market still shows potential. Let's try to get some app ideas based on the kind of apps that are somewhere in the middle in terms of popularity (between 1,000,000 and 100,000,000 downloads):

In [47]:
for app in free_english_android_app:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000+'
                                            or app[5] == '5,000,000+'
                                            or app[5] == '10,000,000+'
                                            or app[5] == '50,000,000+'):
        print(app[0], ':', app[5])

Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
AlReader -any text book reader : 5,000,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
ReadEra – free ebook reader : 1,000,000+
Ebook Reader : 5,000,000+
Read books online : 5,000,000+
eBoox: book reader fb2 epub zip : 1,000,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
Moon+ Reader : 10,000,000+
English-Myanmar Dictionary : 1,000,000+
Golden Dictionary (EN-AR) : 1,000,000+
All Language Translator Free : 1,000,000+
Aldiko Book Reader : 10,000,000+
Dictionary - WordWeb : 5,000,000+
50000 Free eBooks & Free AudioBooks : 5,000,000+
Al-Quran (Free) : 10,000,000+
Al Quran Indonesia : 10,000,000+
Al'Quran Bahasa Indonesia : 10,000,000+
Al Quran Al karim : 1,000,000+
Al Quran : EAlim - Translations & MP3 Offline : 5,000,000+
Koran Read &MP3 30 Juz Offline : 1,000,000+
H

This niche seems to be dominated by software for processing and reading ebooks, as well as various collections of libraries and dictionaries, so it's probably not a good idea to build similar apps since there'll be some significant competition.

We also notice there are quite a few apps built around the book Quran, which suggests that building an app around a popular book can be profitable. It seems that taking a popular book (perhaps a more recent book) and turning it into an app could be profitable for both the Google Play and the App Store markets.

However, it looks like the market is already full of libraries, so we need to add some special features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes on the book, a forum where people can discuss the book, etc.

## Conclusion

In this project, we analyzed data about the App Store and Google Play mobile apps with the goal of recommending an app profile that can be profitable for both markets.

We concluded that taking a popular book (perhaps a more recent book) and turning it into an app could be profitable for both the Google Play and the App Store markets. The markets are already full of libraries, so we need to add some special features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes on the book, a forum where people can discuss the book, etc.